addons/mail/static/src/core/common/thread.js

```diff
@@ -481,6 +481,21 @@ export class Thread extends Component {
                 this.props.thread
             );
         }
+        if (ev.target.tagName === "A") {
+            if (oeModel && oeId) {
+                ev.preventDefault();
+                await this.env.services.action.doAction({
+                    type: "ir.actions.act_window",
+                    res_model: oeModel,
+                    views: [[false, "form"]],
+                    res_id: Number(oeId),
+                });
+                if (!this.env.isSmall) {
+                    this.props.thread.open(true, { autofocus: false });
+                }
+            }
+            return;
+        }
```

Comment:
Not all `a` with `oeModel` and `oeId` should open in place. Only the `#channel` mentions.
To limit appropriately to them, the message of type notification should have a `oeType` and the branching should be on this on `oeType`, similarly to how `highlight` did it above.